src/Microsoft.AspNetCore.Server.Kestrel/Http/Connection.cs

```diff
@@ -152,23 +152,21 @@ public Task StopAsync()
 
         public virtual void Abort()
         {
-            lock (_stateLock)
+            // Frame.Abort calls user code while this method is always
+            // called from a libuv thread.
+            System.Threading.ThreadPool.QueueUserWorkItem(state =>
             {
```

Comment:
Nit: Move `var connection = (Connection)state;` to the top and do `lock (connection._stateLock)`, `connection._connectionState`, etc...